back/spec/acceptance/ideas_spec.rb

```diff
@@ -618,6 +618,14 @@
             expect(json_response[:data][:attributes][:upvotes_count]).to eq 1
           end
 
+          include_context 'when user_blocking duration is 90 days' do
+            example 'Blocked user attempts to create an idea' do
+              @user.update(block_start_at: Time.now)
+              do_request
+              expect(status).to be 401
```

Comment:
Is 401 returned because `User#active?` returns false? In this case, I don't think you need to duplicate the same test for every controller. In my opinion, it's enough to test e.g. `ApplicationPolicy`. Why?
1. Much more code to read and understand when you see this spec.
2. Build time spent to run these tests (slows down our processes and increases bills).
3. Time needed to support (fix and modify) all this duplicated code.

If you really really want to check it in every controller, you can use `shared_examples`, but I'm not sure we need it either.

Reply:
Yes, basically they're nearly all test that `User.active?` is false, so testing `ApplicationPolicy` instead sounds like a great idea. Thanks.

Reply:
Except `ApplicationPolicy` always returns `false` for `create`, `update` and `destroy`, so I can't really test at that level.

The policies that use `active?` are all at the level of specific models, e.g `Idea`.

Reply:
@alexander-cit 

So, I added a shared example to 8 specific model policy specs.

This meant I could remove 21 tests from various acceptance specs, and greatly reduces repetition and verbosity of code.

If you can think of a way to reduce this further (perhaps to one set of tests in a new `application_policy_spec.rb`), then I'd be very interested in this. Maybe we could pair on this, if you think this is possible?

Reply:
Note: we discussed & decided the new tests are good + we should add a simple test of the private `active?` method in `ApplicationPolicy` [Done]